<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1271</width>
    <height>723</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>10</y>
      <width>1281</width>
      <height>691</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="leftMargin">
      <number>20</number>
     </property>
     <property name="topMargin">
      <number>20</number>
     </property>
     <property name="rightMargin">
      <number>20</number>
     </property>
     <property name="bottomMargin">
      <number>20</number>
     </property>
     <item>
      <widget class="QTableWidget" name="processTable">
       <property name="maximumSize">
        <size>
         <width>4000</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Shadow::Raised</enum>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
       </property>
       <property name="columnCount">
        <number>4</number>
       </property>
       <attribute name="horizontalHeaderCascadingSectionResizes">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderCascadingSectionResizes">
        <bool>true</bool>
       </attribute>
       <column>
        <property name="text">
         <string>PID</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Priority</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Arrival Time</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Remaining Burst Time</string>
        </property>
       </column>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_simulationTime">
       <property name="text">
        <string>Simulation Time:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QGraphicsView" name="GanttChartWindow"/>
     </item>
     <item>
      <widget class="QTextEdit" name="textEdit_Statistics">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="layout_input">
       <property name="leftMargin">
        <number>10</number>
       </property>
       <property name="topMargin">
        <number>10</number>
       </property>
       <property name="rightMargin">
        <number>10</number>
       </property>
       <property name="bottomMargin">
        <number>10</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="layout_combobox">
         <item>
          <widget class="QComboBox" name="comboBox">
           <item>
            <property name="text">
             <string>First-Come, First Serve</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Round-Robin</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Priority (Preemptive)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Priority (Non-Preemptive)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Shortest Job First (Preemptive)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Shortest Job First (Non-Preemptive)</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkBox_instantSimulation">
           <property name="text">
            <string>Instant Simulation</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QFormLayout" name="layout_inputForms">
         <property name="labelAlignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
         <property name="formAlignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>5</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="bottomMargin">
          <number>5</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_ArrivalTime">
           <property name="text">
            <string>Arrival Time</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="lineEdit_ArrivalTime"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_BurstTime">
           <property name="text">
            <string>Burst Time</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="lineEdit_BurstTime"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_Priority">
           <property name="text">
            <string>Priority</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="lineEdit_Priority"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="layout_quantum">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>5</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="bottomMargin">
          <number>5</number>
         </property>
         <item>
          <widget class="QLabel" name="label_Quantum">
           <property name="text">
            <string>Time Quantum</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_Quantum"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="layout_buttons">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>5</number>
         </property>
         <property name="rightMargin">
          <number>5</number>
         </property>
         <property name="bottomMargin">
          <number>5</number>
         </property>
         <item>
          <widget class="QPushButton" name="pushButton_InsertProcess">
           <property name="text">
            <string>Insert Process</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_StartSimulation">
           <property name="text">
            <string>Start Simulation</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_ProgramMessages">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1271</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuCPU_Task_Scheduler_Simulator">
    <property name="title">
     <string>CPU Task Scheduler Simulator</string>
    </property>
   </widget>
   <addaction name="menuCPU_Task_Scheduler_Simulator"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
